backend/fregepoc/indexers/models.py

```diff
@@ -60,3 +67,83 @@ def __iter__(self):
     class Meta:
         verbose_name = _("Github Indexer")
         verbose_name_plural = verbose_name
+
+
+class SourceforgeIndexer(BaseIndexer):
+    current_page = models.PositiveIntegerField(
+        _("current page"),
+        default=1,
+        help_text=_("The last visited page."),
+    )
+
+    singlePageProjectsExtractor = SinglePageProjectsExtractor()
+    singleProjectCodeUrlExtractor = SingleProjectCodeUrlExtractor()
+    singleProjectGitLinkExtractor = SingleProjectGitLinkExtractor()
+    singleProjectResponseExtractor = SingleProjectResponseExtractor()
+    singleProjectGitUrlExtractor = SingleProjectGitUrlExtractor()
+
+    def __iter__(self):
+        for i in itertools.count(start=self.current_page):
+            projects = self.singlePageProjectsExtractor.extract(i)
+
+            self.current_page += 1
+            self.save(update_fields=["current_page"])
+
+            if not projects:
+                continue
+
+            for project_name in projects:
+                repos_to_process = []
+
+                single_project_soup = self.singleProjectResponseExtractor.extract(project_name)
+
+                project_code_url = self.singleProjectCodeUrlExtractor.extract(single_project_soup)
+                repo_from_code_url = self.handel_code_url(project_name, project_code_url)
+                if repo_from_code_url is not None:
+                    repos_to_process.append(repo_from_code_url)
+
+                project_git_ulr = self.singleProjectGitUrlExtractor.extract(single_project_soup)
+                repos_to_process.extend(self.handel_git_url(project_name, project_git_ulr))
+
+                Repository.objects.bulk_create(repos_to_process)
+                yield from repos_to_process
+
+    def handel_code_url(self, project_name, project_code_url):
```

Comment:
```suggestion
    def trade_code_url(self, project_name, project_code_url):
```
Please stick to English verbs!

Reply:
Jokes aside, this should be fixed.

```suggestion
    def handle_code_url(self, project_name, project_code_url):
```